### ISSUE TITLE
Regression in 12.8.2: Function IAM auth broken by recent resolver changes

### ISSUE THREAD

**Issue by motravo** [aws-amplify/amplify-category-api#2060]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

9.6.5

### Amplify CLI Version

12.8.2

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No manual changes applicable

### Describe the bug

The resolver changes in https://github.com/aws-amplify/amplify-category-api/pull/1919 moved the definition of `adminRoles` from the auth function request mapping template into the resolver's stash context.

The definition of `adminRoles` was added to the stash context in amplify-graphql-transformer-core here: https://github.com/aws-amplify/amplify-category-api/commit/9297fa5cd#diff-b3129bb53f056565ba11ca27f5846ebb888f0f8bdd643238e2621cd44cd365bdR397

However, it was not added in amplify-graphql-function-transformer/src/graphql-function-transformer.ts. It may also need to be added to amplify-graphql-http-transformer/src/graphql-http-transformer.ts.

The auth function mapping template for a function now expects adminRoles to be present on the stash context, but it is missing, which breaks function auth.

### Expected behavior

`$ctx.stash.adminRoles` should be populated in the resolver stash context if it is expected to be read by the function request mapping template.

### Reproduction steps

Have something like this in schema.graphql:

```
Query {
  myFunction(userId: ID!): SomeObject
    @function(name: "myFunc-${env}")
    @auth(rules: [{ allow: private, provider: iam }])
}
```

Give access to another lambda to call the API.

In amplify 12.7.1, the call is successful. The "before" mapping template for the resolver looks like this:
```
## [Start] Stash resolver specific context.. **
$util.qr($ctx.stash.put("typeName", "Query"))
$util.qr($ctx.stash.put("fieldName", "myFunction"))
$util.qr($ctx.stash.put("authRole", "(trimmed)"))
$util.qr($ctx.stash.put("unauthRole", "(trimmed)"))
{}
## [End] Stash resolver specific context.. **
```
And the QueryMyFunctionAuthFN Request Mapping Template looks like this:
```
## [Start] Field Authorization Steps. **
#set( $isAuthorized = false )
#if( $util.authType() == "IAM Authorization" )
  #set( $adminRoles = ["myfunc1","myfunc2","myfunc3","myadminuser"] )
...
```

In 12.8.2, it fails with "Not Authorized to access myFunction on type Query". Looking at the appsync configuration, the QueryMyFunctionAuthFN Request Mapping Template now looks like this:
```
## [Start] Field Authorization Steps. **
#set( $isAuthorized = false )
#if( $util.authType() == "IAM Authorization" )
  #set( $adminRoles = $ctx.stash.adminRoles )
...
```
but the "before" mapping template is unchanged from 12.7.1 -- $ctx.stash.adminRoles is not defined.

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by ucfangyiwang** [comment]
yeah, get same issue.

**Comment by toddaa** [comment]
Just wanted to add...I'm not using Amplify CLI, but rather the @aws-amplify/graphql-api-construct within AWS CDK.  The fix from #2062 solved my issue where my adminRole was being assumed by an EC2 instance or an IAM user and resulting in "Not Authorized" errors.  After updating everything is working as I thought it would/should originally.
